src/main/java/com/github/dbmdz/solrocr/reader/SourceReader.java

```diff
@@ -36,6 +37,8 @@ public interface SourceReader {
    */
   Section getAsciiSection(int offset) throws IOException;
 
+  Reader getReader() throws IOException;
```

Comment:
I was just looking at the code, maybe we don't need a new API for that? I think all of the I/O in `toCharOffsets` could be expressed through the existing `int read(byte[] dst, int dstOffset, int start, int len)` method. What if we made that part of the `SourceReader` interface and thus public? Then everything in `ExternalUtf8ContentFilterFactory` would be I/O agnostic.

Reply:
Thought some more on it, you were right, `getInputStream()` is the way to go. Problem with using the `read` API is that it's stateless, so we'd need to keep track of the position ourselves. Having a stateful type that does that for us makes our code a lot simpler, and stacking a `java.io.Reader` on top of a `InputStream` is trivial. Thanks!

Reply:
Went for `SeekableByteChannel` in the end, since it's stateful, supports seeking without hacks like `mark` and `reset` and is a subset of the `FileChannel` API we were using already.